src/parser/TurtleParser.h

```diff
@@ -0,0 +1,354 @@
+    auto view = _tok.view();
+    if (ad_utility::startsWith(view, "<")) {
+      auto endPos = view.find_first_of("> \n");
+      if (endPos == string::npos || view[endPos] != '>') {
+        throw ParseException("Iriref");
+      } else {
+        _tok.data().remove_prefix(endPos + 1);
+        _lastParseResult = view.substr(0, endPos + 1);
+        return true;
+      }
+    } else {
+      return false;
+    }
+    // return parseTerminal(_tokens.Iriref);
+  }
+  bool integer() { return parseTerminal(_tokens.Integer); }
+  bool decimal() { return parseTerminal(_tokens.Decimal); }
+  bool doubleParse() { return parseTerminal(_tokens.Double); }
+  bool pnameLN() {
+    _tok.skipWhitespaceAndComments();
+    auto view = _tok.view();
+    auto pos = view.find(":");
+    if (pos == string::npos) {
+      return false;
+    }
+    // these can also be part of a collection etc
+    auto posEnd = view.find_first_of(" \n,;", pos);
+    if (posEnd == string::npos) {
+      // make tests work
+      posEnd = view.size();
+    }
+    // TODO: is it allowed to have no space between triples and the dots? in
+    // this case we have to check something here
+    _activePrefix = view.substr(0, pos);
+    _lastParseResult = view.substr(pos + 1, posEnd - (pos + 1));
+    // we do not remove the whitespace or the ,; since they are needed
+    _tok.data().remove_prefix(posEnd);
+    return true;
+
+    /* if (parseTerminal(_tokens.PnameLN)) {
+       // TODO: better do this with subgroup matching directly
+       auto pos = _lastParseResult.find(':');
+       _activePrefix = _lastParseResult.substr(0, pos);
+       _lastParseResult = _lastParseResult.substr(pos + 1);
+       return true;
+     } else {
+       return false;
+     }
+     */
+  }
+  bool pnameNS() {
+    if (parseTerminal(_tokens.PnameNS)) {
+      // this also includes a ":" which we do not need
+      _activePrefix = _lastParseResult.substr(0, _lastParseResult.size() - 1);
+      _lastParseResult = "";
+      return true;
+    } else {
+      return false;
+    }
+  }
+  bool langtag() { return parseTerminal(_tokens.Langtag); }
+  bool blankNodeLabel() {
+    if (!parseTerminal(_tokens.BlankNodeLabel)) {
+      return false;
+    }
+
+    if (_blankNodeMap.count(_lastParseResult)) {
+      _lastParseResult = _blankNodeMap[_lastParseResult];
+    } else {
+      string blank = createBlankNode();
+      _blankNodeMap[_lastParseResult] = blank;
+      _lastParseResult = blank;
+    }
+    return true;
+  }
+
+  bool anon() {
+    if (!parseTerminal(_tokens.Anon)) {
+      return false;
+    }
+    _lastParseResult = createBlankNode();
+    return true;
+  }
+
+  bool skip(const RE2& reg) {
+    _tok.skipWhitespaceAndComments();
+    return _tok.skip(reg);
+  }
+  bool parseTerminal(const RE2& terminal);
+
+  void emitTriple() {
+    _triples.push_back({_activeSubject, _activePredicate, _lastParseResult});
+  }
+
+  TurtleParserBackupState backupState() const;
```

Comment:
The following 3 methods definitely need method comments. I think for those named after a non-terminal something general should be sufficient